### ISSUE TITLE
Accessibility: should provide a mechanism for specifying a text alternative.

### ISSUE THREAD

**Issue by alice** [Polymer/core-image#13]
Analogous to img[alt].

**Comment by ssorallen** [comment]
Should `alt` stay on the host element or be applied to the placeholder and img elements in the shadow DOM? (Related to #12 as well.)

**Comment by alice** [comment]
`alt` on the host element won't do anything unless you handle it, obviously. And it won't do anything on the placeholder `<div>` either. So what you'll likely need to do is copy it across to the `<img>` if there is one, and use an `aria-label` and `role="img"` on the placeholder `div`. The end result should be that placing screen reader focus on the `<core-image>` speaks something like "<user-provided alt text>, image".

**Comment by ssorallen** [comment]
Given pull request #18 adding a `role="img"` by default to the element, I'm not sure any work needs to be done here. A developer is free to add an `alt` tag to a `<core-image>`, and it should now be correctly interpreted because of the element's default role.

**Comment by alice** [comment]
Unfortunately that's not true for several reasons:
- You're putting the role on an element in the Shadow DOM, not the `<core-image>` itself, so the `alt` attribute would need to be there rather than on the host
- Adding a `role` doesn't actually change the interpretation of any attributes other than ARIA attributes: even if the role went on the host, you'd still need to do the work to interpret the attribute.

I suggest copying the `alt` attribute through to the internal `<img>`, and copying its contents to an `aria-label` on the placeholder div.

**Comment by ssorallen** [comment]
> Adding a `role` doesn't actually change the interpretation of any attributes other than ARIA attributes:

Aha, I didn't realize that.

Was it wrong to add `role="none"` to the `<img>` in the shadow DOM? It seems problematic to have these attributes defined in 3 places: on the host, on the img, and on the placeholder div. Can the host act as the one source of truth for a screen reader?

**Comment by alice** [comment]
Ah, I didn't realise you'd put the role on the host element.

Putting `role="none"` on the `<img>` is unnecessary (but not wrong) - once you put `role="img"` on the host element it becomes a single node in the accessibility tree, so the semantics of any descendant elements are ignored.

So, all you'll need to do is copy the value of `alt` across to an `aria-label` attribute. Alternatively, you could move the `role` and `aria-label` down to the placeholder element, and copy the `alt` attribute down to the `<img>` - benefit of this would be not sprouting attributes on the host, and using native semantics for the `<img>` once it appears, but I'm not sure what this would look like at the point where you transition from the placeholder element to the final `<img>`.
